src/Core/ClassInfo.php

```diff
@@ -364,8 +383,9 @@ public static function table_for_object_field($candidateClass, $fieldName)
      */
     public static function shortName($nameOrObject)
     {
-        $reflection = new ReflectionClass($nameOrObject);
-        return $reflection->getShortName();
+        $name = static::class_name($nameOrObject);
+        $parts = explode('\\', $name);
+        return end($parts);
```

Comment:
looks worse than using `ReflectionClass::getShortName()` but is there a performance gain from this?

Reply:
string manipulation is less expensive than creating a new object.